### PR TITLE
Add comprehensive security and robustness tests and fixes

### PR DESCRIPTION
**Pull request by yjsoon**
## Summary
This PR adds extensive security-focused tests and fixes for input validation, path traversal prevention, email header injection protection, and graceful error handling across the application. The changes harden the codebase against adversarial inputs while improving reliability for edge cases.

### Security Tests & Fixes
- **Project Storage**: Added tests for oversized projects, storage limit enforcement, hostile JSON payloads, prototype pollution via `__proto__`, and filename sanitization in exports
- **Email Abuse Controls**: New module with tests for CRLF header injection prevention, mailbox delimiter validation in sender names, and bounded text validation
- **Email Validation**: Tests for CRLF injection resistance in recipient parsing, deduplication, and RFC-4180 CSV compliance
- **Signed URLs**: Tests for encoded null bytes, backslash traversal, and path normalization edge cases
- **Private File Access**: Tests rejecting namespace directories, null bytes, and non-normalized paths
- **Storage Manager**: Added null byte and double-slash rejection in key validation

### Robustness Improvements
- **PDF Capacity Assessment**: Fixed `estimateTemplateBytes()` to properly validate finite, positive numbers instead of trusting falsy checks
- **PDF Source Loading**: Added `getMaxPdfSourceBytes()` validation to reject non-numeric or invalid environment values; `assertPdfBuffer()` tests for PDF header within first 1024 bytes
- **Coordinate Transform**: Added guard in `scaleFontSize()` to return base size for non-finite or non-positive container widths
- **Session Storage**: Fixed `getSessionAge()` to return `null` for corrupted timestamps instead of `NaN`
- **Account Scope**: Fixed `guardLocalStorageForUser()` to report purge status truthfully even when `setItem` fails
- **Signed URLs**: Enhanced `normalizeGeneratedPdfPath()` to check for null bytes and backslashes after URL decoding
- **CSV Parsing**: Fixed RFC-4180 escaped quote handling (`""` → `"`) in `parseCSVRow()`
- **Middleware**: Fixed `/api` route detection to use exact match instead of `startsWith()`, preventing `/apiary` from being treated as an API route

### Test Coverage Additions
- **Text Measurement**: Comprehensive tests for `splitTextIntoLines()` and `calculateShrinkToFitFontSize()` with mocked canvas context
- **Coordinate Transform**: Tests for coordinate flipping, text box bounds, multiline centering, and font scaling
- **PDF Filenames**: Tests for empty base fallback, non-alphanumeric sanitization, and adversarial naming
- **Rate Limiting**: Tests for window reset timing, array header handling, and missing socket fallbacks
- **Email Quota**: Tests for non-positive recipient counts, unknown users, and oversized requests
- **Table Data Detection**: Tests for CSV edge cases (quoted commas, escaped quotes, duplicate headers, ragged rows)
- **File Upload**: Unskipped previously skipped loading state test
- **Paths**: New test file for `resolvePathWithin()` path confinement primitive
- **Runtime Auth Policy**: Test for whitespace-only environment values

### Implementation Details
- Email header validation rejects `\r`, `\n`, `\0` in all fields; additionally rejects mailbox delimiters (`<>,:;@"\\`) in `senderName` to prevent forged display names
- Path normalization checks both pre- and post-decode for null bytes and backslashes
- PDF buffer validation allows header anywhere in first 1024 bytes per PDF spec
- CSV parsing properly handles RFC-4180 escaped quotes and maintains quoted field integrity
- Storage key validation rejects undefined, non-string, and array inputs
- Timestamp validation uses `Date.parse()` with NaN check to catch corrupted dates

https://claude.ai/code/session_01DnJnC2DA2MVQKJt7PyvELS

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved CSV parsing, including RFC-4180 escaped quotes and expanded CSV edge-case handling.
  * File upload now uses the currently selected file/template settings reliably.
  * Strengthened session validation/age calculation and hardened numeric handling (PDF sizing/capacity, font scaling, byte formatting).
* **Security**
  * Tighter protections for generated and private template paths, signed URL/path normalization, and PDF source/attachment validation.
  * Improved protection against email/header injection and unsafe recipient inputs (including semicolon lists).
* **Tests**
  * Expanded Jest coverage across parsing, storage limits, middleware/API routing, PDF utilities, and rate limiting behavior.

### CODE REVIEW

File: __tests__/hooks/useTableData-detection.test.ts [hunk 1]
```diff
@@ -191,21 +196,146 @@ describe('useTableData Auto-Detection', () => {
...
+
+      await act(async () => {
+        await result.current.loadSessionData('Name,,Email\nJohn,middle,john@example.com', true, true); // CSV mode, headers
+      });
+
+      expect(result.current.tableData).toHaveLength(1);
+      const firstRow = result.current.tableData[0];
+      expect(Object.keys(firstRow)).toEqual(['Name', '_column_1', 'Email']);
+      expect(firstRow['_column_1']).toBe('middle');
+    });
```
[the remark this answers is not in the document]
Comment (yjsoon) [reply __tests__/hooks/useTableData-detection.test.ts:298]: Deliberate decision (confirmed with the maintainer): the parser stays forgiving. This app's primary input path is pasting from Excel/Sheets, where ragged rows and duplicate headers are common in otherwise-usable data — rejecting the paste outright would hurt the main workflow more than the silent normalization does. The pinned tests exist to make the lenient behavior explicit and intentional rather than accidental; if we later add a validation layer (already on the roadmap as "Data Validation" in CLAUDE.md), a non-blocking warning for duplicate headers/ragged rows would be the right shape, not a hard parse error.

The other four findings from this review were valid and are fixed in 3b8c2ed.

---
_Generated by [Claude Code](https://claude.ai/code)_
